File: src/main/java/dk/nmwael/HelloApplication.java
package dk.nmwael;

import io.quarkus.runtime.QuarkusApplication;
import io.quarkus.runtime.annotations.QuarkusMain;
import jakarta.inject.Inject;
import picocli.CommandLine;

@QuarkusMain
public class HelloApplication implements QuarkusApplication {

    private CommandLine.IFactory factory;

    @Inject
    public HelloApplication(CommandLine.IFactory factory){
        this.factory=factory;

    }

    @Override
    public int run(String... args) throws Exception {
         return new CommandLine(this, factory).execute(args);
    }
    
}

File: src/main/kotlin/it/mulders/hello/HelloApplication.kt
package it.mulders.hello

import io.quarkus.runtime.QuarkusApplication
import io.quarkus.runtime.annotations.QuarkusMain
import jakarta.inject.Inject
import picocli.CommandLine

@CommandLine.Command(
    mixinStandardHelpOptions = true,
    name = "hello"
)
@QuarkusMain
class HelloApplication @Inject constructor(val factory: CommandLine.IFactory) : QuarkusApplication {
    override fun run(vararg args: String): Int = CommandLine(this, factory).execute(*args)
}